loopchain/consensus/syncer.py

```diff
@@ -0,0 +1,160 @@
+        self._block_manager = block_manager
+        self.__blockchain = self._block_manager.blockchain
+        self._event_system: 'EventSystem' = event_system
+        self._data_info_other_nodes: Dict[int, list] = {}
+        self._vote_info_other_nodes: Dict[int, list] = {}
+        self._request_history_list = {}
+        self._target_index = 0
+        self._stub_list = []
+        self._max_height_in_nodes = self.__blockchain.block_height
+        self.management_stub()
+
+        if self.__blockchain.last_block:
+            reps_hash = self.__blockchain.get_reps_hash_by_header(self.__blockchain.last_block.header)
+        else:
+            reps_hash = ChannelProperty().crep_root_hash
+        self._rep_targets = self.__blockchain.find_preps_targets_by_roothash(reps_hash)
+
+    def management_stub(self, status="init"):
+        if "update" == status:
+            self._stub_list = []
+            self._target_index = 0
+
+        target_list = self._block_manager.get_target_list()
+        for target in target_list:
+            channel = GRPCHelper().create_client_channel(target)
+            self._stub_list.append(loopchain_pb2_grpc.PeerServiceStub(channel))
+
+    async def sync_start(self):
+        while True:
+            await self._request_block()
+            await self._raise_event()
+
+    async def _request_block(self):
+        def _request(index: int, height: int):
+            peer_stub = self._stub_list[index]
+
+            peer_stub.BlockRequest(loopchain_pb2.PeerHeight(
+                peer=ChannelProperty().peer_target,
+                channel=self._block_manager.channel_name,
+                height=height
+            ), conf.GRPC_TIMEOUT)
+
+        # Sync mode check
+        # LFT is two step process. So It need to wait for a height of at least 2.
+        gap = self._max_height_in_nodes-self.__blockchain.block_height
+        if gap < RAISE_EVENT_INTERVAL:
+            return
+
+        goal = min(conf.CITIZEN_ASYNC_RESULT_MAX_SIZE+1, self._max_height_in_nodes-RAISE_EVENT_INTERVAL)
+        for i in range(1, goal):
+            height = self.__blockchain.block_height + i
+            try:
+                if height in self._request_history_list.keys():
+                    during_request_time = time.time()-self._request_history_list[height][1]
+                    if during_request_time > conf.LFT_SYNC_REQUEST_WAIT:
+                        retry_target_index = (self._request_history_list[height][0] + 1) % len(self._stub_list)
+                        _request(retry_target_index, height)
+                        self._request_history_list[height] = [retry_target_index, time.time()]
+                else:
+                    _request(self._target_index, height)
+                    self._request_history_list[height] = [self._target_index, time.time()]
+                    self._target_index = (self._target_index+1) % len(self._stub_list)
+
```

Comment:
Please minimize try block to describe where you expects the `IndexError` to be raised at.